neqo-transport/src/connection.rs

```diff
@@ -192,11 +195,12 @@ impl Connection {
         protocols: PI,
         local_addr: SocketAddr,
         remote_addr: SocketAddr,
+        external_cert_verification: bool,
     ) -> Res<Connection> {
         let dcid = ConnectionId::generate(CID_LENGTH);
         let mut c = Connection::new(
             Role::Client,
-            Client::new(server_name)?.into(),
+            Client::new(server_name, external_cert_verification)?.into(),
```

Comment:
Any change for the client needs to be mirrored for the server.  A server also gets a peer certificate that might need authentication, even if the certificate is optional (and currently not supported).